src/app/clusters/service-area-server/service-area-delegate.cpp

```diff
@@ -0,0 +1,90 @@
+#include "service-area-delegate.h"
+#include "service-area-server.h"
+
+using namespace chip::app::Clusters::ServiceArea;
+
+bool Delegate::GetSupportedLocationById(uint32_t aLocationId, uint32_t & listIndex, LocationStructureWrapper & aSupportedLocation)
+{
+    listIndex = 0;
+
+    // simple linear iteration to find the location with the desired locationID.
+    while (GetSupportedLocationByIndex(listIndex, aSupportedLocation))
+    {
+        if (aSupportedLocation.locationID == aLocationId)
+        {
+            return true;
+        }
+
+        ++listIndex;
+    }
+
+    return false;
+}
+
+void Delegate::HandleSupportedLocationsUpdated()
+{
+    mInstance->ClearSelectedLocations();
+    mInstance->SetCurrentLocation(DataModel::Nullable<uint32_t>());
```

Comment:
```suggestion
    mInstance->SetCurrentLocation(DataModel::NullNullable);
```
if that compiles, at least.

Reply:
Yeah, it does not compile `error: cannot deduce template arguments for ‘Nullable<...auto...>’`.

Reply:
Fixed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/6cf2ea7077477afd319889a978cae71d5dc0429b).